junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/TestExecutionListenerRegistry.java

```diff
@@ -63,48 +67,117 @@ private void notifyEagerTestExecutionListeners(Consumer<EagerTestExecutionListen
 		this.eagerTestExecutionListeners.forEach(consumer);
 	}
 
-	TestExecutionListener getCompositeTestExecutionListener() {
+	CompositeTestExecutionListener getCompositeTestExecutionListener() {
 		return new CompositeTestExecutionListener();
 	}
 
-	private class CompositeTestExecutionListener implements TestExecutionListener {
+	class CompositeTestExecutionListener implements TestExecutionListener {
+		public List<TestExecutionListener> getAll() {
```

Comment:
Where is this used?

Reply:
in `DefaultLauncher`

`DefaultLauncher`, as the registry to get the `CompositeTestExecutionListener` which is the collection of all listeners and handled the failure too.

Instead of `DefaultLauncher` handing exception one more time, I made this `CompositeTestExecutionListener` visible to `DefaultLuancher` in the same package.

Reply:
I checked out your branch and I could find no usage of this method in `DefaultLauncher` or anywhere else. I think we can make it private again and revert the changes in `DefaultLauncher` or am I missing sth.?

Reply:
Hello @marcphilipp 
The usages of `TestExecutionListener` interface in `DefaultLauncher` are mentioned below.
These are already handled as of `CompositeTestExecutionListener`. But if you don't explicitly use the `Composite` type, it will be hard to reason about it that the exceptions are handled in some internal class.



1. https://github.com/junit-team/junit5/blob/f114fd009778bb4a33cda97a42bdf68d71348221/junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/DefaultLauncher.java#L192


2. https://github.com/junit-team/junit5/blob/f114fd009778bb4a33cda97a42bdf68d71348221/junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/DefaultLauncher.java#L200


3. https://github.com/junit-team/junit5/blob/f114fd009778bb4a33cda97a42bdf68d71348221/junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/DefaultLauncher.java#L207


Let me know if I need to clarify more.


Regards,
Gopinath

Reply:
Thanks, but where is `getAll()` called?

Reply:
Nowhere.

Reply:
Are you suggesting to remove DefaultLauncher Changes, and not handle exception there.

As they are implicitely handled by `CompositeTestExecutionListener`?

Reply:
I may be missing sth. but I don't see any behavioral changes in `DefaultLauncher`. `CompositeTestExecutionListener` was already used there before. And if that now handles exceptions we don't need to change `DefaultLauncher` at all, do we?

Reply:
@marcphilipp You are right, `DefaultLauncher` uses `CompositeTestExecutionListener` instead of Interface `TestExecutionListener` which implicitly means all the exceptions are handled by default.

I was thinking from readability perspective that, If we make the interface method signature to throw an exception then we should explicitly handle the exception in `DefaultLuancher`. Maybe I am thinking more ahead.


We can definitely stop here & keep `DefaultLuancher` as it is.

Let's take one call on this.


Also have a look at below comment too.

Reply:
We're not going to change the interface so let's revert the changes to `DefaultLauncher` and remove this unused method.

Reply:
Sure. I will do that